src/detectors/FHCAL/FHCAL.cc

```diff
@@ -22,30 +22,38 @@ extern "C" {
 
         InitJANAPlugin(app);
 
+        // Make sure digi and reco use the same value
+        decltype(CalorimeterHitDigiConfig::capADC)        HcalEndcapPInsert_capADC = 32768;
+        decltype(CalorimeterHitDigiConfig::dyRangeADC)    HcalEndcapPInsert_dyRangeADC = 200 * dd4hep::MeV;
+        decltype(CalorimeterHitDigiConfig::pedMeanADC)    HcalEndcapPInsert_pedMeanADC = 10;
+        decltype(CalorimeterHitDigiConfig::pedSigmaADC)   HcalEndcapPInsert_pedSigmaADC = 2;
+        decltype(CalorimeterHitDigiConfig::resolutionTDC) HcalEndcapPInsert_resolutionTDC = 10 * dd4hep::picosecond;
+
         app->Add(new JChainMultifactoryGeneratorT<CalorimeterHitDigi_factoryT>(
            "HcalEndcapPInsertRawHits", {"HcalEndcapPInsertHits"}, {"HcalEndcapPInsertRawHits"},
            {
              .eRes = {},
              .tRes = 0.0 * dd4hep::ns,
-             .capADC = 32768,
-             .dyRangeADC = 200 * dd4hep::MeV,
-             .pedMeanADC = 400,
-             .pedSigmaADC = 10,
-             .resolutionTDC = 10 * dd4hep::picosecond,
+             .capADC = HcalEndcapPInsert_capADC,
+             .dyRangeADC = HcalEndcapPInsert_dyRangeADC,
+             .pedMeanADC = HcalEndcapPInsert_pedMeanADC,
+             .pedSigmaADC = HcalEndcapPInsert_pedSigmaADC,
+             .resolutionTDC = HcalEndcapPInsert_resolutionTDC,
              .corrMeanScale = 1.0,
            },
           app   // TODO: Remove me once fixed
         ));
         app->Add(new JChainMultifactoryGeneratorT<CalorimeterHitReco_factoryT>(
           "HcalEndcapPInsertRecHits", {"HcalEndcapPInsertRawHits"}, {"HcalEndcapPInsertRecHits"},
           {
-            .capADC = 32768,
-            .dyRangeADC = 200. * dd4hep::MeV,
-            .pedMeanADC = 400,
-            .pedSigmaADC = 10.,
-            .resolutionTDC = 10 * dd4hep::picosecond,
+            .capADC = HcalEndcapPInsert_capADC,
+            .dyRangeADC = HcalEndcapPInsert_dyRangeADC,
+            .pedMeanADC = HcalEndcapPInsert_pedMeanADC,
+            .pedSigmaADC = HcalEndcapPInsert_pedSigmaADC,
+            .resolutionTDC = HcalEndcapPInsert_resolutionTDC,
             .thresholdFactor = 0.,
-            .thresholdValue = -100.,
+            .thresholdValue = 41.0, // 0.25 MeV --> adc = 50 + 0.25 / 200 * 32768 = 91
```

Comment:
Could you please explain this line to me? Why do we use 0.25 MeV and 50 in adc =50 + ... ? And the threshold value is just the 0.25 / 200 * 32768 portion, or should it be 91? Sorry for the naive questions -- I'm not that familiar with these numbers.

Reply:
The 50 is wrong (I looked at the wrong entry in the spreadhseet), but yes, for the threshold it just gets subtracted anyway, I just spelled it out here, because that's the adc calculation. In summary, the 41 remains correct. Let me plug in the other pedestalMean, just to be consistent.

Reply:
@rymilton Done. Thanks for catching the typo (only wrong in the comments, but I simplified that comment to match what you pointed out)

Reply:
Thanks for the explanation! That looks good to me then.